### PR TITLE
chore: mini nits from ics29 audit

### PR DESCRIPTION
**Pull request by seantking**
## Description

There is a call for adding a var declaration in the issue as well but I wasn't sure of the reasoning. 

closes: #1325 

---

Before we can merge this PR, please make sure that all the following items have been
checked off. If any of the checklist items are not applicable, please leave them but
write a little note why.

- [ ] Targeted PR against correct branch (see [CONTRIBUTING.md](https://github.com/cosmos/ibc-go/blob/master/CONTRIBUTING.md#pr-targeting))
- [ ] Linked to Github issue with discussion and accepted design OR link to spec that describes this work.
- [ ] Code follows the [module structure standards](https://github.com/cosmos/cosmos-sdk/blob/master/docs/building-modules/structure.md).
- [ ] Wrote unit and integration [tests](https://github.com/cosmos/ibc-go/blob/master/CONTRIBUTING.md#testing)
- [ ] Updated relevant documentation (`docs/`) or specification (`x/<module>/spec/`)
- [ ] Added relevant `godoc` [comments](https://blog.golang.org/godoc-documenting-go-code).
- [ ] Added a relevant changelog entry to the `Unreleased` section in `CHANGELOG.md`
- [ ] Re-reviewed `Files changed` in the Github PR explorer
- [ ] Review `Codecov Report` in the comment section below once CI passes

### CODE REVIEW

File: modules/apps/29-fee/keeper/escrow.go [hunk 1]
```diff
@@ -165,6 +166,7 @@ func (k Keeper) distributeFee(ctx sdk.Context, receiver, refundAccAddress sdk.Ac
 		// then attempt to refund the fee to the original sender
 		err := k.bankKeeper.SendCoinsFromModuleToAccount(cacheCtx, types.ModuleName, refundAccAddress, fee)
 		if err != nil {
+			k.Logger(ctx).Error("Error refunding fee to the original sender", "Refund Address:", refundAccAddress, "Fee:", fee)
```
Comment (crodriguezvega) [inline modules/apps/29-fee/keeper/escrow.go:169]: Sorry, this is super nit, but I think in other places where we log we do not use capital letters or colons...
```suggestion
			k.Logger(ctx).Error("error refunding fee to the original sender", "refund address", refundAccAddress, "fee", fee)
```

File: modules/apps/29-fee/keeper/msg_server.go [hunk 2]
```diff
@@ -21,7 +21,7 @@ func (k Keeper) RegisterCounterpartyAddress(goCtx context.Context, msg *types.Ms
 		ctx, msg.Address, msg.CounterpartyAddress, msg.ChannelId,
 	)
 
-	k.Logger(ctx).Info("Registering counterparty address for relayer.", "Address:", msg.Address, "Counterparty Address:", msg.CounterpartyAddress)
+	k.Logger(ctx).Info("Registering counterparty address for relayer.", "Address:", msg.Address, "Counterparty Address:", msg.CounterpartyAddress, "Channel:", msg.ChannelId)
```
Comment (crodriguezvega) [inline modules/apps/29-fee/keeper/msg_server.go:24]:
```suggestion
	k.Logger(ctx).Info("registering counterparty address for relayer", "address", msg.Address, "counterparty address", msg.CounterpartyAddress, "channel", msg.ChannelId)
```

**Review by crodriguezvega — APPROVED** [review]
I left some nits of these nits... :) 

The reasoning to do `var` declaration [here](https://github.com/cosmos/ibc-go/blob/main/modules/apps/29-fee/ibc_middleware.go#L235) was to just be consistent with the rest of the codebase.

File: modules/apps/29-fee/keeper/escrow.go [hunk 3]
```diff
@@ -165,6 +166,7 @@ func (k Keeper) distributeFee(ctx sdk.Context, receiver, refundAccAddress sdk.Ac
 		// then attempt to refund the fee to the original sender
 		err := k.bankKeeper.SendCoinsFromModuleToAccount(cacheCtx, types.ModuleName, refundAccAddress, fee)
 		if err != nil {
+			k.Logger(ctx).Error("error refunding fee to the original sender", "refund address", refundAccAddress, "fee", fee)
```
Comment (colin-axner) [inline modules/apps/29-fee/keeper/escrow.go:169]: I think we should log the first failed distribution. I think it is very unlikely refunding the fee fails, but trying to distribute to a relayer could definitely fail. It'd be useful for that relayer to notice and see that the fee got refunded after the failed distribution

**Review by colin-axner — COMMENTED** [review]
LGTM, it'd be great if you could add the var declaration as well

**Review by damiannolan — APPROVED** [review]
Agree with @colin-axner re. moving logging up one level of scope. 

About the var declaration, it's misleading in the issue and linked code because of recent changes. The code to be changed is the `ack` initialisation [here](https://github.com/cosmos/ibc-go/blob/main/modules/apps/29-fee/ibc_middleware.go#L251). 

Removing the `new()` func and using the `var` declaration
```
ack := new(types.IncentivizedAcknowledgement)
if err := types.ModuleCdc.UnmarshalJSON(acknowledgement, ack); err != nil {
	return sdkerrors.Wrapf(err, "cannot unmarshal ICS-29 incentivized packet acknowledgement: %v", ack)
}
```

**Pull request merged**